### ISSUE TITLE
Encoding failed error when Exporting to WebM

### ISSUE THREAD

**Issue by Mr-Anonymous** [pascalbaljetmedia/laravel-ffmpeg#58]
I have installed `laravel-ffmpeg` today on my Laravel 5.3.

First I was getting `Encoding failed` error when I was exporting the video to mp4:
```
FFMpeg::fromDisk('videos')
              ->open('steve_howe.mkv')
              ->export()
              ->inFormat(new FFMpeg\Format\Video\X264)
              ->save('short_steve.mp4') 
```

After reading the topics here, I fixed it by adding `libmp3lame` audio codec and it worked:
```
FFMpeg::fromDisk('videos')
              ->open('steve_howe.mkv')
              ->export()
              ->inFormat(new FFMpeg\Format\Video\X264('libmp3lame'))
              ->save('short_steve.mp4') 
```

However, I am at the moment unable to convert the video to WebM format. I have this at the moment:
```
FFMpeg::fromDisk('videos')
              ->open('steve_howe.mkv')
              ->export()
              ->inFormat(new FFMpeg\Format\Video\WebM)
              ->save('short_steve.webm') 
```
I tried adding `libvorbis` codec to it but I still keep getting `Encoding failed` error. How can I convert the videos to WebM? Do I need to add any specific audio codec to make this work?

Thank you for your help.

- Neel.

**Comment by Mr-Anonymous** [comment]
Ok, here is an update on this. I reinstalled `ffmpeg` again using the flags `--with-libvpx --with-libvorbis`. After this, the WebM videos are being converted. However, the quality is extremely poor. Its so poor that you can't even view the video and it just looks like a static video.
```
brew reinstall ffmpeg --with-libvpx --with-libvorbis -v
```
```
./configure --prefix=/usr/local/Cellar/ffmpeg/3.4.1 --enable-shared --enable-pthreads --enable-version3 --enable-hardcoded-tables --enable-avresample --cc=clang --host-cflags= --host-ldflags= --enable-gpl --enable-libmp3lame --enable-libvorbis --enable-libvpx --enable-libx264 --enable-libxvid --enable-opencl --enable-videotoolbox --disable-lzma
```
```
FFMpeg::fromDisk('videos')
              ->open('steve_howe.mkv')
              ->export()
              ->inFormat(new FFMpeg\Format\Video\WebM)
              ->save('short_steve.webm') 
```
Why am I having so much issues with WebM conversion? Any advise please?

**Comment by Mr-Anonymous** [comment]
I finally got it resolved. Installing just `libvorbis` was not enough. I also had to use `libvpx-vp9` video codec instead of the default one. Without this, the video was very pixelated and green. I resolved it by doing this:

Reinstalled ffmpeg with these dependencies:
```
brew reinstall ffmpeg --with-fdk-aac --with-libass --with-sdl2 --with-x265 --with-freetype --with-libvorbis --with-libvpx --with-opus --with-theora
```
Used `libvorbis` and `libvpx-vp9` codecs like this:
```
new \FFMpeg\Format\Video\WebM('libvorbis', 'libvpx-vp9')
```

Finally after the above updates, the videos are now converted to WebM properly.

Neel.
